src/cdk/table/table.ts

```diff
@@ -232,19 +268,25 @@ export class CdkTable<T> implements CollectionViewer, OnInit, AfterContentChecke
     if (this._headerRowDef) {
       this._headerRowDefChanged = true;
     }
+
+    // If the table has a footer row definition defined as part of its content, flag this as a
+    // header row def change so that the content check will render the header row.
+    if (this._footerRowDef) {
+      this._footerRowDefChanged = true;
+    }
```

Comment:
Could this check (and the header one above) be replaced with 
```ts
this._footerRowDefChanged = !!this._footerRowDef;
```
Or should it not be always set to false when `_footerRowDef` is undefined?